src/main/scala/net/snowflake/spark/snowflake/Parameters.scala

```diff
@@ -698,6 +703,9 @@ object Parameters {
     def supportAWSStageEndPoint: Boolean = {
       isTrue(parameters.getOrElse(PARAM_INTERNAL_SUPPORT_AWS_STAGE_END_POINT, "true"))
     }
+    def supportShareConnection: Boolean = {
+      isTrue(parameters.getOrElse(PARAM_SUPPORT_SHARE_CONNECTION, "true"))
```

Comment:
Enable it by default will be a behavior change. I think we should disable it by default.

Reply:
I prefer to regard this is a performance optimization instead of a behaviour change.
With this change, we expect the customer didn't see any change except it is faster.

It's like `ENABLE_FIX_<JIRA>` in snowflake. It is introduced just in order to give the user a way to disable in case of regression.